tests/integrations/raftstore/test_replica_read.rs

```diff
@@ -309,3 +309,68 @@ fn test_read_index_out_of_order() {
     assert!(resp2.recv_timeout(Duration::from_secs(1)).is_ok());
     assert!(resp1.recv_timeout(Duration::from_secs(1)).is_ok());
 }
+
+#[test]
+fn test_split_isolation() {
+    let mut cluster = new_node_cluster(0, 2);
+    // Use long election timeout and short lease.
+    configure_for_hibernate(&mut cluster);
+    configure_for_lease_read(&mut cluster, Some(200), Some(10));
+    cluster.cfg.raft_store.raft_log_gc_count_limit = 11;
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+
+    let rid = cluster.run_conf_change();
+    pd_client.must_add_peer(rid, new_learner_peer(2, 2));
+    cluster.must_put(b"k1", b"v1");
+    cluster.must_put(b"k2", b"v2");
+    must_get_equal(&cluster.get_engine(2), b"k1", b"v1");
+    must_get_equal(&cluster.get_engine(2), b"k2", b"v2");
+    cluster.must_transfer_leader(1, new_peer(1, 1));
+
+    cluster.stop_node(2);
+    // Split region into ['', 'k2') and ['k2', '')
+    let r1 = cluster.get_region(b"k2");
+    cluster.must_split(&r1, b"k2");
+    let idx = cluster.truncated_state(1, 1).get_index();
+    // Trigger a log compaction, so the left region ['', 'k2'] cannot created through split cmd.
+    for i in 0..cluster.cfg.raft_store.raft_log_gc_count_limit * 2 {
```

Comment:
```suggestion
    for i in 2..cluster.cfg.raft_store.raft_log_gc_count_limit * 2 {
```
Maybe should start from `k2` so all put will send to the region 1